dotcom-rendering/src/web/components/Callout/FileUpload.tsx

```diff
@@ -1,31 +1,79 @@
 import { css } from '@emotion/react';
-import { space, text, textSans } from '@guardian/source-foundations';
-import { useState } from 'react';
-import type { CampaignFieldFile } from './../../../types/content';
-import { stringifyFileBase64 } from './../../lib/stringifyFileBase64';
+import type { SerializedStyles } from '@emotion/react';
+import {
+	neutral,
+	palette,
+	remHeight,
+	space,
+	textSans,
+	visuallyHidden,
+} from '@guardian/source-foundations';
+import React, { useState } from 'react';
+import type { CampaignFieldType } from 'src/types/content';
+import { decidePalette } from 'src/web/lib/decidePalette';
+import { stringifyFileBase64 } from 'src/web/lib/stringifyFileBase64';
 import { FieldLabel } from './FieldLabel';
 
-const fileUploadInputStyles = css`
-	padding-top: 10px;
-	padding-bottom: 10px;
-`;
-
 const errorMessagesStyles = css`
 	padding-top: ${space[2]}px;
-	color: ${text.error};
+	color: ${palette.error};
 	${textSans.small({ fontWeight: 'bold' })};
 `;
 
+const uploadStyles = css`
+	display: block;
+`;
+
+const textStyles = css`
+	${textSans.small()};
+	color: ${neutral[46]};
+	display: inline-block;
+`;
+
 type Props = {
-	formField: CampaignFieldFile;
+	format: ArticleFormat;
+	formField: CampaignFieldType;
 	formData: { [key in string]: any };
 	setFormData: React.Dispatch<React.SetStateAction<{ [x: string]: any }>>;
+	validationErrors: { [key in string]: string };
 };
 
-export const FileUpload = ({ formField, formData, setFormData }: Props) => {
+const customUpload = (format: ArticleFormat): SerializedStyles => css`
+	${textSans.small()};
+	color: ${decidePalette(format).text.richLink};
+	border: 1.5px solid ${decidePalette(format).text.richLink};
+	display: inline-flex;
+	justify-content: space-between;
+	align-items: center;
+	box-sizing: border-box;
+	background: transparent;
+	cursor: pointer;
+	transition: all 0.3s ease-in-out 0s;
+	text-decoration: none;
+	white-space: nowrap;
+	height: ${remHeight.ctaXsmall}rem;
+	min-height: ${remHeight.ctaXsmall}rem;
+	padding: ${space[3]}px;
+	margin: ${space[3]}px ${space[3]}px 0px 0px;
+	border-radius: ${remHeight.ctaMedium}rem;
+	${textSans.medium({ fontWeight: 'bold' })};
+	width: fit-content;
+`;
+
+export const FileUpload = ({
+	formField,
+	format,
+	formData,
+	setFormData,
+}: Props) => {
+	const [chosenFile, setChosenFile] = useState<null | string>();
 	const [error, setError] = useState('');
+
+	const getFileName = (filepath?: string): string =>
+		filepath?.split(/(\\|\/)/g).pop() ?? '';
+
 	const onSelectFile = async (event: React.ChangeEvent<HTMLInputElement>) => {
-		if (event.target.files && event.target.files[0]) {
+		if (event.target.files?.[0]) {
```

Comment:
The && operator seemed unnecessary here, hence I removed it

Reply:
We even have a proposal to enforce this in the future: #6715